src/hyp3_srg/time_series.py

```diff
@@ -4,56 +4,42 @@
 
 import argparse
 import logging
+import re
 import shutil
 from os import mkdir
 from pathlib import Path
 from secrets import token_hex
 from shutil import copyfile
 from typing import Iterable, Optional
 
-from boto3 import client
 from hyp3lib.aws import upload_file_to_s3
 from hyp3lib.fetch import download_file as download_from_http
 
 from hyp3_srg import dem, utils
 
 
-S3 = client('s3')
 log = logging.getLogger(__name__)
 
 
-def get_s3_args(uri: str, dest_dir: Optional[Path] = None) -> None:
-    """Retrieve the arguments for downloading from an S3 bucket
+def get_gslc_uris_from_s3(bucket: str, prefix: str = '') -> list[str]:
+    """Retrieve granule (zip files) uris from the given s3 bucket and prefix.
 
     Args:
-        uri: URI of the file to download
-        dest_dir: the directory to place the downloaded file in
+        bucket: the s3 bucket name
+        prefix: the path after the bucket and before the file
 
     Returns:
-        bucket: the s3 bucket to download from
-        key: the path to the file following the s3 bucket
-        out_path: the destination path of the file to download
+        uris: a list of uris to the zip files
     """
-    if dest_dir is None:
-        dest_dir = Path.cwd()
+    bucket = bucket.replace('s3:', '').replace('/', '')
+    res = utils.s3_list_objects(bucket, prefix)
 
-    simple_s3_uri = Path(uri.replace('s3://', ''))
-    bucket = simple_s3_uri.parts[0]
-    key = '/'.join(simple_s3_uri.parts[1:])
-    out_path = dest_dir / simple_s3_uri.parts[-1]
-    return bucket, key, out_path
+    def is_valid_key(key):
+        return (key.endswith('.zip') or key.endswith('.geo')) and re.search('S1[AB]_IW_RAW', key.split('/')[-1])
```

Comment:
Would any of the outputs ever end with `.geo`?

Although, now that the GSLCs are uploaded to a separate sub-prefix, we could just remove the `is_valid_key` check entirely because the prefix should *only* contain GSLCs.